scim-core/src/main/java/org/apache/directory/scim/core/repository/PatchHandlerImpl.java

```diff
@@ -279,11 +300,14 @@ public <T extends ScimResource> void applyMultiValue(T source, Map<String, Objec
           throw new UnsupportedFilterException("Attribute cannot be added, only comparison expressions are supported when the existing item does not exist.");
         }
         AttributeComparisonExpression comparisonExpression = (AttributeComparisonExpression) valuePathExpression.getAttributeExpression();
+        checkPrimary(subAttributeName, items, value);
 
-      items.add(Map.of(
+        items.add(new HashMap<>(Map.of(
```

Comment:
Does this need to be a HashMap?

Reply:
No I suppose it doesn't. Thing is `Map.of` creates an Immutable map by default, which later causes trouble when we try to call `put`... I was trying to solve it when I came across [this](https://stackoverflow.com/questions/54963337/create-mutable-collections-map-of-and-list-of)....
But I am open to any suggestions of course

Reply:
```suggestion
        // Add a new mutable map
        items.add(new HashMap<>(Map.of(
```